This patch will allow you to forward a message simply by
tapping on the text of a message in the chat view. It does
not interfere with the current attachment-tapping behavior.
Tapping an attached image will still prompt for a save, etc.

USAGE: Open the Messaging application, tap a chat thread,
then tap an individual message.  If the message has an
attachment, make sure you tap on the text of the message,
rather than the attachment.
Index: /usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
===================================================================
--- .orig/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
+++ /usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
@@ -1443,6 +1443,10 @@ var ChatviewAssistant = Class.create({
 			this.controller.stageController.pushScene('mmsTextAttachment', filePath); 			
 			return;		
 		}		
+		
+		if (!mmsImageTarget && !mmsVideoTarget && !mmsVcardTarget && !mmsVcalTarget) {
+			MessagingMojoService.forwardMessage(this.controller, event.item.messageText);
+		}
 				
 		MessagingUtils.simpleListClick(this.controller.get(event.originalEvent.target), "chatRow", function(targetRow){
 			var messageData = {
Index: /usr/palm/applications/com.palm.app.messaging/app/models/messaging-luna-service.js
===================================================================
--- .orig/usr/palm/applications/com.palm.app.messaging/app/models/messaging-luna-service.js
+++ /usr/palm/applications/com.palm.app.messaging/app/models/messaging-luna-service.js
@@ -5,6 +5,19 @@ var MessagingMojoService = {
   accountsIdentifier: 'palm://com.palm.accounts',
   displayIdentifier: 'palm://com.palm.display/control',
   telephonyIdentifier: 'palm://com.palm.telephony',
+  forwardIdentifier: 'palm://com.palm.applicationManager',
+  
+  forwardMessage: function(sceneController,messageText) {
+  	return sceneController.serviceRequest(MessagingMojoService.forwardIdentifier,{
+  		method: 'launch',
+  		parameters: {
+  			id: 'com.palm.app.messaging',
+  			params: {
+  				messageText: 'FWD: '+messageText
+  			}
+  		}
+  	});
+  },
 
   launchMessaging: function(sceneController,params) {  
 	return sceneController.serviceRequest("palm://com.palm.applicationManager", {
Index: /usr/palm/applications/com.palm.app.messaging/app/views/chatview/chatList-row.html
===================================================================
--- .orig/usr/palm/applications/com.palm.app.messaging/app/views/chatview/chatList-row.html
+++ /usr/palm/applications/com.palm.app.messaging/app/views/chatview/chatList-row.html
@@ -1,4 +1,4 @@
-<div id="chat_#{messageId}" class="palm-row chat-entry chatRow #{css_class}  #{accountClass}" errorCode="#{statusErrorCode}" status="#{status}" messageId="#{messageId}" flags="#{flags}" messageType="#{messageType}">
+<div id="chat_#{messageId}" class="palm-row chat-entry chatRow #{css_class}  #{accountClass}" errorCode="#{statusErrorCode}" status="#{status}" messageId="#{messageId}" flags="#{flags}" messageType="#{messageType}" x-mojo-tap-highlight="momentary">
 	<div class="palm-row-wrapper">
 		<div class="chat-balloon message #{messageTextClass} message-container">
 			<div class="chat-balloon-wrapper">
